### PR TITLE
[Client consolidation] Migrate from legacy elasticsearch client to opensearch-js client in `osd-opensearch-archiver` package

### PR DESCRIPTION
**Pull request by manasvinibs**
### Description
In this PR, we are replacing legacy elasticsearch client package which is used to interact with Opensearch APIs with opensearch-js client inside `osd-opensearch-archiver` package as a pre-req for version decoupling of Opensearch Dashboards and Opensearch.
Reference https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3900

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4041

### Check List

- [X] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-opensearch-archiver/src/lib/docs/index_doc_records_stream.ts [hunk 1]
```diff
@@ -58,8 +58,8 @@ export function createIndexDocRecordsStream(
       );
     });
 
-    const resp = await client.bulk({ requestTimeout: 2 * 60 * 1000, body });
-    if (resp.errors) {
+    const resp = await client.bulk({ body }, { requestTimeout: 2 * 60 * 1000 });
+    if (resp.body.errors) {
```
Comment (AMoo-Miki) [inline packages/osd-opensearch-archiver/src/lib/docs/index_doc_records_stream.ts:62]: We should also throw if `resp.body` is `undefined`

**Review by joshuarrrr — APPROVED** [review]
One codecov check failing due to lack of patch coverage. It would be nice to raise the test coverage with this refactor where possible, but it's not a blocker for me.

**Review by kavilla — DISMISSED** [review]

**Comment by joshuarrrr** [comment]
all tests passed before CHANGELOG resolution.

**Review by joshuarrrr — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Comment by kavilla** [comment]
Link checker failing on this: https://sass-lang.com/blog/libsass-is-deprecated. I think it's good to merge. Being that is primarily used within functional tests. Do we want to consider backporting this?

**Comment by manasvinibs** [comment]
> Link checker failing on this: https://sass-lang.com/blog/libsass-is-deprecated. I think it's good to merge. Being that is primarily used within functional tests. Do we want to consider backporting this?

I've added the backport label. I think it should be safe to backport this change. Let me know if we don't want to.

File: packages/osd-opensearch-archiver/src/lib/docs/index_doc_records_stream.test.ts [hunk 2]
```diff
@@ -63,11 +63,12 @@ describe('opensearchArchiver: createIndexDocRecordsStream()', () => {
     const client = createStubClient([
       async (name, params) => {
         expect(name).to.be('bulk');
-        expect(params).to.eql({
-          body: recordsToBulkBody(records),
-          requestTimeout: 120000,
```
[the remark this answers is not in the document]
Comment (manasvinibs) [reply packages/osd-opensearch-archiver/src/lib/docs/index_doc_records_stream.test.ts:68]: Earlier bulk API would take `requestTimeout` input as part of `BulkRequest` param, but with new bulk API, `requestTimeout` is not part of bulkRequest Param interface (renamed/re-typed to timeout), instead it is used as part of `TransportRequestOptions`. 
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4142/files/0fb11084c023cf65a27a4cdbdd626c70576e213a#diff-a2f5251433aa1701470ba8eec10215aff6149564b512d7be3e97aa3a1b967190R61

https://github.com/opensearch-project/opensearch-js/blob/e3efb225e5880fe4f5a4403dbc10eaa5e2173e31/api/opensearch_dashboards.d.ts#L91

[BulkRequest](https://github.com/opensearch-project/opensearch-js/blob/main/api/types.d.ts#L69C11-L69C33)

**Pull request merged**